### ISSUE TITLE
Changes we can make now we depend on Astropy 4.0+

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#4000]
This is a list of things which we can change once we depend on Astropy 4.0+. This is a follow on from #3154

## Visualization

- [x] (fixed in https://github.com/sunpy/sunpy/pull/4621) Line plotting in `ArrayAnimatorWCS` is conditional on Astropy 4.0, if it's our minimum requirement we can drop all the conditional stuff and expect it to be there.

## Coordinates

- [x] (Fixed in #4609) Remove explicit transformations of coordinates prior to calling `.plot_coord()` (e.g., [here](https://docs.sunpy.org/en/latest/generated/gallery/units_and_coordinates/STEREO_SECCHI_starfield.html) and [here](https://docs.sunpy.org/en/latest/generated/gallery/saving_and_loading_data/coordinates_in_asdf.html)) because the transformation will be performed automatically.
- [x] (Fixed in #4577) Remove custom `size` property of `SunPyBaseCoordinateFrame` because that workaround is no longer needed (see astropy/astropy#10295).
- [x] (Fixed in #4584) Change `apparent_longitude()` and `apparent_latitude()` to use `GeocentricTrueEcliptic` (no change in accuracy) (requires Astropy 4.0.2, see astropy/astropy#10129)

**Comment by nikita-0209** [comment]
Since Astropy 4.0 has been released, can I begin working on this issue? I am a novice.

**Comment by nikita-0209** [comment]
Oh alright. I'll look for another issue

**Comment by mridullpandey** [comment]
hi, @Cadair  may I please take this up.

**Comment by nabobalis** [comment]
@mridullpandey we still do not depend on astropy 4.0 only, I would suggest another issue.

**Comment by mridullpandey** [comment]
thanks

**Comment by Cadair** [comment]
I still think it's too early to do this, but following the pattern of incrementing our astropy dep by one each release, we can make these changes on master in a month or so.

**Comment by kathivashkiv** [comment]
Are changes listed above relevant for now? (After #3918 discussion)

**Comment by nabobalis** [comment]
Yes.

**Comment by kathivashkiv** [comment]
About one of them, "delete size property": 
1) there is one mention of this property along with SunPyBaseCoordinateFrame, in metaframe.py:
https://github.com/sunpy/sunpy/blob/master/sunpy/coordinates/metaframes.py#L184
Should this check be deleted after size property removal, or should it stay there for another reason (different than described)?
2) I've run tests for package='coordinates' and sunpy.self_test() after property removal. Which tests need to be performed additionally (or instead) before PR?

**Comment by nabobalis** [comment]
> About one of them, "delete size property":
> 
> 1. there is one mention of this property along with SunPyBaseCoordinateFrame, in metaframe.py:
>    [https://github.com/sunpy/sunpy/blob/master/sunpy/coordinates/metaframes.py#L184](https://github.com/sunpy/sunpy/blob/master/sunpy/coordinates/metaframes.py?rgh-link-date=2020-10-16T14%3A28%3A55Z#L184)
>    Should this check be deleted after size property removal, or should it stay there for another reason (different than described)?
> 2. I've run tests for package='coordinates' and sunpy.self_test() after property removal. Which tests need to be performed additionally (or instead) before PR?

1. I think it means this: https://github.com/sunpy/sunpy/blob/master/sunpy/coordinates/frames.py#L144 

2. Sounds like enough.

**Comment by ayshih** [comment]
> 1. Should this check be deleted after size property removal, or should it stay there for another reason (different than described)?

Thanks for spotting that!  Go ahead and remove the check, and I'll figure out how to confirm that it's fine.  The existing test suite won't tell us.

**Comment by dstansby** [comment]
For 

-  Remove custom size property of SunPyBaseCoordinateFrame because that workaround is no longer needed (see astropy/astropy#10295).

shouldn't we add a deprecation warning before changing this behaviour? (posting here to avoid clogging up the above PR)

**Comment by Cadair** [comment]
Maybe, I am erring on the side of no, because there is no way to opt of of a deprecation warning which says "in the next version of sunpy I will be an exception".

**Comment by ayshih** [comment]
In principle, we could have a deprecation period, but I'm against doing so for this case because I'd like to eliminate discrepant behavior between SunPy frames and Astropy frames as quickly as possible.  (This discrepant behavior is my fault, but that doesn't make it any less discrepant.)  The discrepant behavior applies only to calling `.size` on a frame with no associated data; I'm fairly confident that there are no real users that would be affected by this change.

**Comment by dstansby** [comment]
Sounds good, just wanted to double check 😄

**Comment by dstansby** [comment]
That looks like everything ticked off the list 🎉
